language/functional_tests/tests/testsuite/generics/coin_wrapper.mvir

```diff
@@ -0,0 +1,109 @@
+module M {
+    import 0x0.LibraCoin;
+    resource Foo<T: resource> { x: T }
+    resource Bar { f: Self.Foo<LibraCoin.T> }
+    resource Baz { f: Self.Foo<Self.Inner>}
+    resource Inner { b: bytearray }
+
+    public bar(c: LibraCoin.T) {
```

Comment:
Since you are writing some meaningful logic here (a coin wrapper), I'd suggest you give the module, functions, structs & variables meaningful names. It'll help people understand what the code is doing and make it easier to see what's being tested.

Other than that the test seems good to me. (We will add a lot more tests later & refactor existing ones, so no need to worry too much right now.)